### DISCUSSION TITLE
`getModel` gone?

### DISCUSSION THREAD

**Discussion by SirEndii** [cc-tweaked/CC-Tweaked#1158] [Q&A] [answered]
I currently port AP to 1.19, and I saw that the `ITurtleUpgrade` misses the `getModel` method
But since we rotate the turtle model at actions, we would need something like this

**Comment by SquidDev** [comment, marked as answer]
Ahh, this is [documented in `TurtleUpgradeSerialiser`](https://github.com/cc-tweaked/CC-Tweaked/blob/mc-1.19.x/src/main/java/dan200/computercraft/api/turtle/TurtleUpgradeSerialiser.java) but not `ITurtleUpgrade`.

The method has just moved to [a separate interface](https://github.com/cc-tweaked/CC-Tweaked/blob/mc-1.19.x/src/main/java/dan200/computercraft/api/client/turtle/TurtleUpgradeModeller.java).

**Reply by SirEndii** [reply]
Thank you good sir

**Reply by SquidDev** [reply]
NP. Aware the interface is a little clunky, so I guess any feedback welcome.

**Reply by SirEndii** [reply]
All fixed now, the  turtles render again and the model can be rotated
At the first sight the new system looks a bit useless, but after using it I can tell that it isn't.  It's easy to use and perfect for our use cases
